paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/pipeline/cdc/util/DataConvertTest.java

```diff
@@ -0,0 +1,157 @@
+                        .physicalColumn(
+                                "timestamp_with_precision_9",
+                                org.apache.flink.cdc.common.types.DataTypes.TIMESTAMP(9))
+                        .physicalColumn(
+                                "timestamp_ltz",
+                                org.apache.flink.cdc.common.types.DataTypes.TIMESTAMP_LTZ())
+                        .physicalColumn(
+                                "timestamp_ltz_with_precision_3",
+                                org.apache.flink.cdc.common.types.DataTypes.TIMESTAMP_LTZ(3))
+                        .physicalColumn(
+                                "timestamp_ltz_with_precision_6",
+                                org.apache.flink.cdc.common.types.DataTypes.TIMESTAMP_LTZ(6))
+                        .physicalColumn(
+                                "timestamp_ltz_with_precision_9",
+                                org.apache.flink.cdc.common.types.DataTypes.TIMESTAMP_LTZ(9))
+                        .primaryKey("pk_string")
+                        .partitionKey("boolean")
+                        .build();
+        TableId tableId = TableId.tableId("testDatabase", "testTable");
+        BinaryRecordDataGenerator recordDataGenerator =
+                new BinaryRecordDataGenerator(
+                        fullTypesSchema.getColumnDataTypes().toArray(new DataType[0]));
+        Object[] testData =
+                new Object[] {
+                    BinaryStringData.fromString("pk_string"),
+                    true,
+                    new byte[] {1, 2, 3},
+                    new byte[] {4, 5, 6},
+                    new byte[] {7, 8, 9},
+                    (byte) 1,
+                    (short) 2,
+                    3,
+                    4L,
+                    5.1f,
+                    6.2,
+                    DecimalData.fromBigDecimal(new BigDecimal("7.123"), 6, 3),
+                    BinaryStringData.fromString("test1"),
+                    BinaryStringData.fromString("test2"),
+                    BinaryStringData.fromString("test3"),
+                    DateData.fromEpochDay(1000),
+                    TimeData.fromMillisOfDay(200),
+                    TimeData.fromMillisOfDay(300).toMillisOfDay(),
+                    TimestampData.fromMillis(100, 1),
+                    TimestampData.fromMillis(200, 2),
+                    TimestampData.fromMillis(300, 3),
+                    TimestampData.fromMillis(400, 4),
+                    LocalZonedTimestampData.fromEpochMillis(300, 3),
+                    LocalZonedTimestampData.fromEpochMillis(400, 4),
+                    LocalZonedTimestampData.fromEpochMillis(500, 5),
+                    LocalZonedTimestampData.fromEpochMillis(600, 6),
+                };
+        org.apache.flink.cdc.common.event.DataChangeEvent dataChangeEvent =
+                DataChangeEvent.insertEvent(tableId, recordDataGenerator.generate(testData));
+
+        Assertions.assertEquals(
```

Comment:
Let's also verify the converted data in the middle phase, after the original CDC data is converted to paimon data and before it's converted back.